spec/unit/resource/service_spec.rb

```diff
@@ -124,6 +124,22 @@
     end.to raise_error(ArgumentError)
   end
 
+  it "should accept true for recursive" do
+    @resource.recursive true
```

Comment:
These should all be `@resource.options`, and you're expecting them to take an Array or a String, rather than a bool.